subsys/bluetooth/host/hci_core.c

```diff
@@ -1225,19 +1318,34 @@ static void enh_conn_complete(struct bt_hci_evt_le_enh_conn_complete *evt)
 			 * Handle cancellation of outgoing connection attempt.
 			 */
 			if (conn->err == BT_HCI_ERR_UNKNOWN_CONN_ID) {
+#if !defined(CONFIG_BT_WHITELIST)
 				/* We notify before checking autoconnect flag
 				 * as application may choose to change it from
 				 * callback.
 				 */
 				bt_conn_set_state(conn, BT_CONN_DISCONNECTED);
 
-#if !defined(CONFIG_BT_WHITELIST)
 				/* Check if device is marked for autoconnect. */
 				if (atomic_test_bit(conn->flags,
 						    BT_CONN_AUTO_CONNECT)) {
+					/* Restart passive scanner for device */
 					bt_conn_set_state(conn,
 							  BT_CONN_CONNECT_SCAN);
 				}
+#else
+				if (atomic_test_bit(conn->flags,
+						    BT_CONN_AUTO_CONNECT)) {
+
+					/* Restart whitelist initiator after
+					 * RPA timeout.
+					 */
+					bt_le_create_conn(conn);
+				} else {
+					/* Create connection canceled by timeout
+					 */
+					bt_conn_set_state(conn,
+							  BT_CONN_DISCONNECTED);
+				}
 #endif /* !defined(CONFIG_BT_WHITELIST) */
```

Comment:
Can this be converted to use `IS_ENABLED()` I don't see any obvious dependency problems.

Reply:
It has an indentation problem, and IS_ENABLED introduces a new indentation.
I can combine some of the outer if statements and use IS_ENABLED.

Reply:
The deep indentation is a problem already in itself - it's an indication that this function needs some redesign - possibly by refactoring into some smaller helper functions or rethinking the function flow. Either way, I'm fine leaving this to a later PR.

Reply:
Pushed a refactor commit to this. If you don't like it I can remove it again.